aten/src/ATen/SavedTensorHooks.h

```diff
@@ -1,20 +1,49 @@
 #pragma once
 
 #include <c10/macros/Export.h>
+#include <c10/util/Optional.h>
 #include <c10/util/python_stub.h>
 #include <stack>
+#include <string>
 
 #include <utility>
 
 namespace at {
 
+namespace impl {
+
+struct TORCH_API SavedTensorDefaultHooksTLS {
+  // PyObject is defined in c10/util/python_stub.h
+  std::stack<std::pair<PyObject*, PyObject*>> stack;
+
+  // See NOTE: [Disabling SavedTensorDefaultHooks]
+  c10::optional<std::string> disabled_error_message;
+  bool is_disabled = false;
+};
+
+} // namespace impl
+
 struct TORCH_API SavedTensorDefaultHooks {
   static void push_hooks(PyObject* pack_hook, PyObject* unpack_hook);
   static void pop_hooks();
   static std::pair<PyObject*, PyObject*> get_hooks();
-  static void enable();
+  static void enable(); // NB: has nothing to do with disable() below.
```

Comment:
Should we rename this to global_enable?

Reply:
Might need to name this something else, since we need to split set_disabled to set_disabled and set_enabled.